napari/_qt/qt_resources/_icons.py

```diff
@@ -0,0 +1,360 @@
+
+    This constructs a temporary directory and builds a .qrc file as described
+    in https://doc.qt.io/qt-5/resources.html
+
+    Parameters
+    ----------
+    xmls : Iterable[Tuple[str, str]]
+        An iterable of ``(alias, xml)`` pairs, where `alias` is the name that
+        you want to use to access the icon in the Qt Resource system (such as
+        QSS), and `xml` is the *raw* SVG text (as read from a file, perhaps
+        pre-colored using one of the below functions).
+        The output of :func:`generate_colorized_svgs` is a suitable input for
+        `xmls`.
+    prefix : str, optional
+        A prefix to use when accessing these resources.  For instance, if
+        ``prefix='theme'``, and one of the aliases in `xmls` is
+        `"my_icon.svg"`, then you could access the compiled resources at
+        `:/theme/my_icon.svg`. by default ''
+
+    Yields
+    ------
+    Iterator[str]
+        Yields the *name* of the temporary generated .qrc file, which can be
+        used as input to :func:`compile_qrc`.
+    """
+    with TemporaryDirectory() as tdir_name:
+        tmp_dir = Path(tdir_name)
+
+        # create the colorized SVGs
+        files = []
+        for alias, xml in xmls:
+            path = alias.replace("/", "_")
+            (tmp_dir / path).write_text(xml)
+            files.append(FILE_T.format(alias=alias, path=path))
+
+        # write the QRC file
+        res_file = tmp_dir / 'res.qrc'
+        res_file.write_text(QRC_TEMPLATE.format(prefix, '\n'.join(files)))
+        yield str(res_file)
+
+
+def generate_colorized_svgs(
+    svg_paths: Iterable[Union[str, Path]],
+    colors: Iterable[Union[str, Tuple[str, str]]],
+    opacities: Iterable[float] = (1.0,),
+    theme_override: Optional[Dict[str, str]] = None,
+) -> Iterator[Tuple[str, str]]:
+    """Helper function to generate colorized SVGs.
+
+    This is a generator that yields tuples of ``(alias, icon_xml)`` for every
+    combination (cartesian product) of `svg_path`, `color`, and `opacity`
+    provided. It can be used as input to :func:`_temporary_qrc_file`.
+
+    Parameters
+    ----------
+    svg_paths : Iterable[Union[str, Path]]
+        An iterable of paths to svg files
+    colors : Iterable[Union[str, Tuple[str, str]]]
+        An iterable of colors.  Every icon will be generated in every color. If
+        a `color` item is a string, it should be valid svg color style. Items
+        may also be a 2-tuple of strings, in which case the first item should
+        be an available theme name
+        (:func:`~napari.utils.theme.available_themes`), and the second item
+        should be a key in the theme (:func:`~napari.utils.theme.get_theme`),
+    opacities : Iterable[float], optional
+        An iterable of opacities to generate, by default (1.0,) Opacities less
+        than one can be accessed in qss with the opacity as a percentage
+        suffix, e.g.: `my_svg_50.svg` for opacity 0.5.
```

Comment:
```suggestion
        suffix, e.g.: ``my_svg_50.svg`` for opacity 0.5.
```

Reply:
i think I disagree here... all of these double backticks seems to be directly inline with numpy docstrings advice on "inline display" here: https://numpydoc.readthedocs.io/en/latest/format.html#other-points-to-keep-in-mind

Reply:
they state that "Variable, module, function, and class names should be written between single back-ticks (`numpy`)." ... but not _everything_...

Reply:
yes, but you've put single backticks... and numpy says:

>  For inline display use double backticks

So we agree ??

Reply:
lol... 😂  we agree :)